hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4716,6 +4753,38 @@ func (r *HostedClusterReconciler) discoverKubevirtClusterClient(ctx context.Cont
 	return cluster, nil
 }
 
+func (r *HostedClusterReconciler) reconcileSREMetricsConfig(ctx context.Context, createOrUpdate upsert.CreateOrUpdateFN, cpNamespace string) error {
+	log := ctrl.LoggerFrom(ctx)
+	if r.MetricsSet != metrics.MetricsSetSRE {
+		return nil
+	}
+	log.Info("Reconciling SRE metrics configuration")
+	cm := metrics.SREMetricsSetConfigurationConfigMap(r.OperatorNamespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(cm), cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("SRE configuration does not exist")
+			return nil
+		}
+		return fmt.Errorf("failed to get SRE configuration configmap: %w", err)
+	}
+	currentMetricsSetConfigHash := metrics.SREMetricsSetConfigHash(cm)
+	if currentMetricsSetConfigHash != r.SREConfigHash {
+		// Only load a new config if configuration content has changed
+		if err := metrics.LoadSREMetricsSetConfigurationFromConfigMap(cm); err != nil {
```

Comment:
because the singleton variable in support/metrics/sets.go needs to be updated with the latest SRE config. From the HO we create the service monitor for the cpo, so it needs to be available for this process as well in case you want to update that config.